### DISCUSSION TITLE
Avoid hardcoding passwords

### DISCUSSION THREAD

**Discussion by tomkennes** [argoproj-labs/argocd-notifications#378] [Q&A]
Hi there,

Just a question, and potentially a feature request. 

Is there a way to make use of secrets within the notifications configmaps?

I would like to embed a Bearer token, which should be possible through the headers, but I would like to avoid having to put in the secret. Feels a bit odd to just put it there without any protection.

Thanks!

**Comment by leoskyrocker** [comment]
As far as I understand, you can refer to secrets via `$my-secret` in the config map. Separately, you'd store `my-secret` under the secret `argocd-notifications-secret` (default secret object name). Once you do that, you'll be able to reference it in the configmap.
